src/Autofac/Features/AttributeFilters/MetadataFilterAttribute.cs

```diff
@@ -160,9 +161,11 @@ public override object ResolveParameter(ParameterInfo parameter, IComponentConte
             var elementType = GetElementType(parameter.ParameterType);
             var hasMany = elementType != parameter.ParameterType;
 
-            return hasMany
+            value = hasMany
                 ? FilterAllMethod.MakeGenericMethod(elementType).Invoke(null, new[] { context, Key, Value })
                 : FilterOneMethod.MakeGenericMethod(elementType).Invoke(null, new[] { context, Key, Value });
+
+            return true;
```

Comment:
If this is a `Try` method... is there any point at which it'd return false? Or is it still "success or throw?"